### PR TITLE
feature/FOUR-8264: Log Creation, cancellation, and completion of REQUEST

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps
Create, Cancelled or Completed a Request

## How to Test
- Create a Request
- Go to admin an edit user
- Select tab security logs
- The same when you delete an complete a Request

## Related Tickets & Packages
[- Link to any related FOUR tickets, PRDs, or packages
](https://processmaker.atlassian.net/browse/FOUR-8264)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

## Enviroment
ci:deploy

### CODE REVIEW

File: ProcessMaker/Events/RequestAction.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    
+    /**
+     * Return event data 
+     */
+    public function getData(): array
+    {
+        $actionAt = '';
+        
+        switch ($this->action) {
```
Comment (fagubla) [inline ProcessMaker/Events/RequestAction.php:37]: Instead use switch, use `array_search`

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 2]
```diff
@@ -1277,6 +1279,8 @@ public function triggerStartEvent(Process $process, Request $request)
             ], 422);
         }
 
+        event(new RequestAction($processRequest, RequestAction::ACTIONS[0]));
```
Comment (fagubla) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:1282]: Hardcode 0, send the label

File: ProcessMaker/Observers/ProcessRequestObserver.php [hunk 3]
```diff
@@ -45,6 +46,7 @@ public function deleting(ProcessRequest $request)
     public function saved(ProcessRequest $request)
     {
         if ($request->status === 'COMPLETED') {
+        event(new RequestAction($request, RequestAction::ACTIONS[2]));
```
Comment (fagubla) [inline ProcessMaker/Observers/ProcessRequestObserver.php:49]: use the correct indentation

**Review by fagubla — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 4]
```diff
@@ -1277,6 +1279,8 @@ public function triggerStartEvent(Process $process, Request $request)
             ], 422);
         }
 
+        event(new RequestAction($processRequest, 'CREATED'));
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php]: event(new RequestAction($processRequest, RequestAction::ACTION_CREATED));

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

File: ProcessMaker/Events/RequestAction.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+use ProcessMaker\Contracts\SecurityLogEventInterface;
+use ProcessMaker\Models\Process;
+use ProcessMaker\Models\ProcessRequest;
+
+class RequestAction implements SecurityLogEventInterface
+{
+    use Dispatchable;
+
+    const ACTION_CREATED = 'CREATED';
+    const ACTION_CANCELLED = 'CANCELLED';
```
Comment (fagubla) [inline ProcessMaker/Events/RequestAction.php:16]: misspelled put **CANCELED**

**Review by fagubla — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Review by fagubla — APPROVED** [review]

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Events/RequestAction.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+use ProcessMaker\Contracts\SecurityLogEventInterface;
+use ProcessMaker\Models\Process;
+use ProcessMaker\Models\ProcessRequest;
+
+class RequestAction implements SecurityLogEventInterface
+{
+    use Dispatchable;
+
+    const ACTION_CREATED = 'CREATED';
+    const ACTION_CANCELED = 'CANCELED';
```
Comment (pmPaulis) [inline ProcessMaker/Events/RequestAction.php:16]: alphabetical order

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Review by pmPaulis — APPROVED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-b0d031d831.eng.processmaker.net

**Pull request merged**